### PR TITLE
Use testcontainers to run `redis-server`

### PR DESCRIPTION
**Pull request by AlexanderYastrebov**
Mitigates build failures caused by package manager redirects during `redis-server` installation.
Note that change adds many dependencies but they are only used for tests and do not end up in the build.

Signed-off-by: Alexander Yastrebov <alexander.yastrebov@zalando.de>

### CODE REVIEW

**Comment by AlexanderYastrebov** [comment]
If that works we can also use it for etcd
https://github.com/zalando/skipper/blob/100ef1b584304d1e17f33d07cf8952ded84252cb/etcd/etcdtest/mocketcd.go#L45-L47

File: net/redistest/redistest.go [hunk 1]
```diff
@@ -2,57 +2,71 @@ package redistest
...
+	ctx, cancel := context.WithTimeout(context.Background(), 20*time.Second)
+	defer cancel()
 
-	cmd := exec.CommandContext(ctx, "redis-server", args...) // #nosec
-	if err := cmd.Start(); err != nil {
+	container, err := testcontainers.GenericContainer(ctx, testcontainers.GenericContainerRequest{
+		ContainerRequest: testcontainers.ContainerRequest{
+			Image:        "redis:6-alpine",
+			Cmd:          args,
+			ExposedPorts: []string{"6379/tcp"},
```
Comment (szuecs) [inline net/redistest/redistest.go:33]: Does a test container gets it's own IP?

Comment (AlexanderYastrebov) [reply net/redistest/redistest.go:33]: It gets a mapped port on localhost that we obtain using .Endpoint below

Comment (szuecs) [reply net/redistest/redistest.go:33]: sounds like we could test sharding with this too, but we need to change the port for the next ones :)

**Comment by szuecs** [comment]
Do you also want to add etcd?

**Comment by AlexanderYastrebov** [comment]
> Do you also want to add etcd?

I thought about doing it in separate PR

File: delivery.yaml [hunk 2]
```diff
@@ -8,8 +8,6 @@ pipeline:
   commands:
   - desc: build-push
     cmd: |
-      apt-get update  -o Acquire::http::AllowRedirect=false
-      apt-get install -o Acquire::http::AllowRedirect=false -y bzr redis-server jq
```
Comment (AlexanderYastrebov) [inline delivery.yaml:12]: I don't see `bzr` used anywhere but I think this would break a Makefile that depends on `jq`

**Comment by aryszka** [comment]
:+1:

**Pull request merged**